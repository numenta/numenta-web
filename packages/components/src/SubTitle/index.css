/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

._heading {
  composes: lh-title from 'tachyons';
  composes: mt1 from 'tachyons';
  composes: w-60-ns from 'tachyons';
  composes: w-100 from 'tachyons';
}


.h2 {
  composes: b--black-20 from 'tachyons';
  composes: bb from 'tachyons';
  composes: dark-gray from 'tachyons';
  composes: f3 from 'tachyons';
  composes: fw5 from 'tachyons';
  composes: mb2 from 'tachyons';
  composes: pb1 from 'tachyons';
  composes: _heading;
  border-bottom-style: dotted;
}
.h3 {
  composes: f4 from 'tachyons';
  composes: fw6 from 'tachyons';
  composes: fw5-ns from 'tachyons';
  composes: mid-gray from 'tachyons';
  composes: mb1 from 'tachyons';
  composes: _heading;
}
.h4 {
  composes: f4 from 'tachyons';
  composes: fw4 from 'tachyons';
  composes: mb1 from 'tachyons';
  composes: gray from 'tachyons';
  composes: _heading;
}
.h5 {
  composes: f5 from 'tachyons';
  composes: fw3 from 'tachyons';
  composes: silver from 'tachyons';
  composes: mb1 from 'tachyons';
  composes: _heading;
}
.h6 {
  composes: f5 from 'tachyons';
  composes: fw3 from 'tachyons';
  composes: light-silver from 'tachyons';
  composes: mb1 from 'tachyons';
  composes: _heading;
}
