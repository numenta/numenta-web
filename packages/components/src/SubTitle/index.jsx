// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import React from 'react'

import styles from './index.css'


const SubTitle = ({children, level}) => {
  const Heading = `h${level}`

  return (
    <Heading className={styles[Heading]}>
      {children}
    </Heading>
  )
}

SubTitle.propTypes = {
  children: React.PropTypes.node.isRequired,
  level: React.PropTypes.number,
}

SubTitle.defaultProps = {
  level: 2,
}

export default SubTitle
