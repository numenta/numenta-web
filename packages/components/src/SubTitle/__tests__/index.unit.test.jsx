import React from 'react'
import renderer from 'react-test-renderer'

import SubTitle from '../../SubTitle'


describe('SubTitle React component', () => {

  it('Renders Heading Level 2 correctly', () => {
    const component = renderer.create(
      <SubTitle>
        Level 2 Title
      </SubTitle>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Heading Level 3 correctly', () => {
    const component = renderer.create(
      <SubTitle level={3}>
        Level 3 Title
      </SubTitle>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Heading Level 4 correctly', () => {
    const component = renderer.create(
      <SubTitle level={4}>
        Level 4 Title
      </SubTitle>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Heading Level 5 correctly', () => {
    const component = renderer.create(
      <SubTitle level={5}>
        Level 5 Title
      </SubTitle>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Heading Level 6 correctly', () => {
    const component = renderer.create(
      <SubTitle level={6}>
        Level 6 Title
      </SubTitle>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
