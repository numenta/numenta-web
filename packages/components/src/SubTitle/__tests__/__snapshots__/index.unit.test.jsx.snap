exports[`SubTitle React component Renders Heading Level 2 correctly 1`] = `
<h2
  className="h2">
  Level 2 Title
</h2>
`;

exports[`SubTitle React component Renders Heading Level 3 correctly 1`] = `
<h3
  className="h3">
  Level 3 Title
</h3>
`;

exports[`SubTitle React component Renders Heading Level 4 correctly 1`] = `
<h4
  className="h4">
  Level 4 Title
</h4>
`;

exports[`SubTitle React component Renders Heading Level 5 correctly 1`] = `
<h5
  className="h5">
  Level 5 Title
</h5>
`;

exports[`SubTitle React component Renders Heading Level 6 correctly 1`] = `
<h6
  className="h6">
  Level 6 Title
</h6>
`;
